src/Tutor.Web/Controllers/Learners/EnrollmentController.cs

```diff
@@ -24,18 +25,16 @@ public class EnrollmentController : BaseApiController
     }
 
     [HttpGet]
-    public ActionResult<List<CourseDto>> GetEnrolledCourses()
+    public ActionResult<PagedResult<CourseDto>> GetEnrolledCourses([FromQuery] int page, [FromQuery] int pageSize)
```

Comment:
I should have called yesterday to clarify my comment. The thinking for the PagedResult should be:

- If the controller is returning a PagedResult and the client is actually using it meaningfully, we should orient the service and repository to return a PagedResult.
- If it is working with a list, and for some reason it is returning a PagedResult that is always equal to the full list (and content of the DB), just to get the "Count" field, I think we should think about reworking the client to derive the count.

We can always avoid sending the query params to get everything, so it's not a big issue. What do you think?

Reply:
Yes, I wasn't sure about the approach either.

It is reasonable to have only list response where needed, for example Learner Course navigation in the sidebar. But then, service would have to have to methods, "GetAll" and "GetAllPaged". Because of the implementation of the extension method, which returns everything if "page" and "pageSize" are not present, the methods would be pretty much the same.

Should we have somewhat overlapping endpoints for that? Is there a best practice for API design? Should everything be paged?

Reply:
Also, regarding "GetPaged": because it is async method, where should it be awaited?
Found two ways in the current code:

- in the service: https://github.com/Clean-CaDET/tutor/blob/master/src/Tutor.Core/UseCases/Management/Groups/LearnerGroupService.cs#L27-L29
- in the repo: https://github.com/Clean-CaDET/tutor/blob/master/src/Tutor.Infrastructure/Database/Repositories/Knowledge/CourseDatabaseRepository.cs#L15-L17

Reply:
Probably in the service, but pick the more common option.